x/delayedack/keeper/rollapp_packet.go

```diff
@@ -77,6 +79,25 @@ func (k Keeper) GetPendingPacketsByAddress(ctx sdk.Context, receiver string) ([]
 	return packets, nil
 }
 
+// GetPendingPacketsByAddressPaginated retrieves rollapp packets from the KVStore by their receiver with pagination.
+func (k Keeper) GetPendingPacketsByAddressPaginated(ctx sdk.Context, receiver string, pageReq *query.PageRequest) (packets []commontypes.RollappPacket, pageResp *query.PageResponse, err error) {
+	_, pageResp, err = collcompat.CollectionPaginate(ctx, k.pendingPacketsByAddress, pageReq,
+		func(key collections.Pair[string, []byte], _ collections.NoValue) (bool, error) {
+			packet, err := k.GetRollappPacket(ctx, string(key.K2()))
+			if err != nil {
+				return true, err
+			}
+			packets = append(packets, *packet)
+			return false, nil
+		}, collcompat.WithCollectionPaginationPairPrefix[string, []byte](receiver),
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return packets, pageResp, nil
+}
```

Comment:
this could be simpler. you can directly return `commontypes.RollappPacket` in `transformFunc` while paginating.
```suggestion
// GetPendingPacketsByAddressPaginated retrieves rollapp packets from the KVStore by their receiver with pagination.
func (k Keeper) GetPendingPacketsByAddressPaginated(ctx sdk.Context, receiver string, pageReq *query.PageRequest) ([]commontypes.RollappPacket, *query.PageResponse, error) {
	return collcompat.CollectionPaginate(ctx, k.pendingPacketsByAddress, pageReq,
		func(key collections.Pair[string, []byte], _ collections.NoValue) (commontypes.RollappPacket, error) {
			packet, err := k.GetRollappPacket(ctx, string(key.K2()))
			return *packet, err
		}, collcompat.WithCollectionPaginationPairPrefix[string, []byte](receiver),
	)
}
```

Reply:
```
packet, err := k.GetRollappPacket(ctx, string(key.K2()))
return *packet, err
```

`packet` could be nil though